### ISSUE TITLE
Ability to turn off prettier

### ISSUE THREAD

**Issue by flippidippi** [nrwl/nx#7421]
Add the ability to turn off prettier, specifically in generators. Currently, prettier is installed automatically with npm 7/8, which there is no way to completely turn it off.

## Description

For teams that dislike prettier, it would be nice to be able to turn it completely off.

## Motivation

Prettier can cause headaches in larger teams and unnecessary code changes to review.

## Suggested Implementation

Take into account the `.prettierignore` for all prettier tasks.

## Alternate Implementations

Either add a setting in `nx.json`, or change prettier to be `optionalDependencies` and checking if installed before using.

**Comment by adlion** [comment]
On VS Code you can easily disable prettier on a specific workspace by just going on extensions and disabling it.
I am sure other IDE have something similar.

**Comment by flippidippi** [comment]
Reopen. Still an issue.

**Comment by nawilson** [comment]
Hi,

This is still an issue in NX version 13.8.0.
With the prettier packages removed (prettier, eslint-config-prettier, eslint-plugin-prettier), eslint fails with the following error: 

```
UnhandledPromiseRejectionWarning:` Error: Failed to load config "prettier" to extend from.
Referenced from: .../node_modules/@nrwl/eslint-plugin-nx/src/index.js
    at configInvalidError (.../node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2596:9)
```

I had to remove prettier from the `extends` attribute of the default eslint configs in:
```
@nrwl/eslint-plugin-nx/src/configs/javascript.js
@nrwl/eslint-plugin-nx/src/configs/typescript.js
```

**Comment by root0x** [comment]
I have also encountered this same issue. I would like to not use prettier but keep eslint. Can it be possible to have prettier be optional?

**Comment by Rugshtyne** [comment]
This is an issue. We are not using prettier though the eslint-plugin-nx wants to extend the config of it. Please decouple the extension in the configs.

**Comment by johnameyer** [comment]
@FrozenPandaz can we reopen this issue? Also seeing
```console
Failed to load config "prettier" to extend from.
Referenced from: @nrwl/eslint-plugin-nx/src/index.js

**Comment by B-R-Bender** [comment]
I can see .prettierignore in the root of the repo
why just not to add

> /apps
> /libs

into it while this issue still open?

**Comment by tukusejssirs** [comment]
Until this issue is fixed, one can simply merge `eslint` config from the following files from `@nrwl/eslint-plugin-nx` into their root `eslint` config (of course, without `prettier` extension):

- [`javascript.ts`](https://github.com/nrwl/nx/blob/master/packages/eslint-plugin-nx/src/configs/javascript.ts);
- [`typescript.ts`](https://github.com/nrwl/nx/blob/master/packages/eslint-plugin-nx/src/configs/typescript.ts).

Don’t forget to remove these `extends` from your `eslint` config:
- `plugin:@nrwl/nx/typescript`;
- `plugin:@nrwl/nx/javascript`.

**Comment by danielwong2268** [comment]
hey @meeroslav , this PR has added about 6+ seconds of lint overhead to our current project. It's particularly bad because it's a mono repo, and this is doing an `npm ls` 4 times, which is costly.

Overall we have 28 total libs/apps in our NX monorepo, each which incurs a 6 second hit. now our lint build has gone from ~40 seconds to 7+ minutes. this will not scale well for us as we add more.

Is this something we can look into?

**Comment by meeroslav** [comment]
Hey @danielwong2268,

The performance impact you are mentioning has been fixed a few weeks ago in this PR: https://github.com/nrwl/nx/pull/11603 and landed in `v14.5.7`.

Please update to the latest version of nx.

**Comment by danielwong2268** [comment]
👍  thank you
